### PR TITLE
TF-7749 Update Policy Sets API documentation

### PR DESCRIPTION
**Pull request by Netra2104**
### What

Should be merged alongside [431](https://github.com/hashicorp/terraform-docs-common/pull/431)

This PR updates the Policy Sets API documentation:

Adds a "Detach a Policy Set from exclusions" endpoint
Adds a "Attach a Policy Set from exclusions" endpoint

Updates Policy Sets API endpoints to take into account updating and returning excluded workspaces associated with a policy set.

### Why

We want to give the TFC users the ability to exclude a subset of workspaces from a policy set that is enforced globally or on a project(s). Some customers need exceptions for individual workspaces and without this ability, they would need to individually add each workspace to a policy set. These new endpoints support this work.

----------

### Merge Checklist
_If items do not apply to your changes, add (N/A) and mark them as complete._

#### Pull Request
- [x] One or more labels describe the type of change (e.g. clarification) and associated product (e.g. tfc).
- [x] Description links to related pull requests or issues, if any.

#### Content
- [x] Redirects have been added to `website/redirects.js` for moved, renamed, or deleted pages.
- [x] API documentation and the API Changelog have been updated. 
- [x] Links to related content where appropriate (e.g., API endpoints, permissions, etc.).
- [x] Pages with related content are updated and link to this content when appropriate.
- [x] Sidebar navigation files have been updated for added, deleted, reordered, or renamed pages.
- [x] New pages have metadata (page name and description) at the top.
- [x] New images are 2048 px wide. They have HashiCorp standard annotation color (#F92672) and format (rectangle with rounded corners), blurred sensitive details (e.g. credentials, usernames, user icons), and descriptive alt text in the markdown for accessibility.
- [x] New code blocks have the correct syntax and line breaks to eliminate horizontal scroll bars.
- [x] UI elements (button names, page names, etc.) are bolded.
- [x] The Vercel website preview successfully deployed.

#### Reviews
- [x] I or someone else reviewed the content for technical accuracy.
- [x] I or someone else reviewed the content for typos, punctuation, spelling, and grammar.

### CODE REVIEW

File: website/docs/cloud-docs/api-docs/policies.mdx [hunk 1]
```diff
@@ -40,7 +40,7 @@ description: >-
...
 
-[Policy sets](/terraform/cloud-docs/policy-enforcement/manage-policy-sets) are collections of policies that you can apply globally or to specific [projects](/terraform/cloud-docs/workspaces/organize-workspaces-with-projects) and workspaces, in your organization. For each run in the selected workspaces, Terraform Cloud checks the Terraform plan against the policy set and displays the results in the UI.
+[Policy sets](/terraform/cloud-docs/policy-enforcement/manage-policy-sets) are collections of policies that you can apply globally or to specific [projects](/terraform/cloud-docs/workspaces/organize-workspaces-with-projects), workspaces and exclusions, in your organization. For each run in the selected workspaces, Terraform Cloud checks the Terraform plan against the policy set and displays the results in the UI.
```
Comment (ker-an) [inline website/docs/cloud-docs/api-docs/policies.mdx:43]: I don't think we need this change 😸 Here we are talking about how you can apply policy sets and it's only globally, to workspaces and to projects.

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 2]
```diff
@@ -1124,7 +1251,8 @@ The following relationships may be present in various responses for policy sets:
 | `organization`    | The organization associated with the specified policy set.                                                                                                                                     |
 | `policies`        | Individually managed policies which are associated with the policy set.                                                                                                                        |
 | `projects`        | The project this policy set applies to.                                                                                                                                                  |
```
Comment (ker-an) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:1253]: Something we missed previously:

```suggestion
| `projects`        | The projects this policy set applies to.                                                                                                                                                  |
```

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 3]
```diff
@@ -1124,7 +1251,8 @@ The following relationships may be present in various responses for policy sets:
...
-| `workspaces`      | The workspaces to which the policy set applies.                                                                                                                                                |
+| `workspaces`      | The workspaces to which the policy set applies. 
+| `exclusions`      | The excluded workspaces to which the policy set applies.                                                                                                                                                |
```
Comment (ker-an) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:1255]:
```suggestion
| `workspaces`      | The workspaces this policy set applies to. 
| `exclusions`      | The workspaces excluded from to this policy set's enforcement.                                                                                                                                                |
```

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ---
 page_title: Policy Sets - API Docs - Terraform Cloud
 description: >-
-  Use the `/policy-sets` endpoint to manage groupings of Sentinel and OPA policies. List, show, create, and update policy sets and policy set versions, and attach and detach policies from workspaces and projects using the HTTP API.
+  Use the `/policy-sets` endpoint to manage groupings of Sentinel and OPA policies. List, show, create, and update policy sets and policy set versions, and attach and detach policies from workspaces, projects, and exclusions using the HTTP API.
```
Comment (ker-an) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:4]: I wonder if there is a better way to word this whole thing 🤔 I'll think about this more! It's getting a bit unruly and finding it hard to understand. Rose might have some thoughts!

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 5]
```diff
@@ -568,7 +589,8 @@ Properties without a default value are required.
...
+| `data.relationships.workspaces`               | array\[object] | (previous value) | An array of references to workspaces that the policy set should be assigned to. Sending an empty array clears all workspace assignments. You can only specify this attribute when `data.attributes.global` is `false`.                                                                                       | 
+| `data.relationships.exclusions`               | array\[object] | (previous value) | An array of references to excluded workspaces that the policy set should be assigned to. Sending an empty array clears all exclusions assignments. You can only specify this attribute when `data.attributes.global` is `false`.                                                                                      |                                                                                                                                          |
```
Comment (ker-an) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:593]:
```suggestion
| `data.relationships.exclusions`               | array\[object] | (previous value) | An array of references to workspaces that the policy set should not be assigned to. Sending an empty array clears all exclusions assignments. You can only specify this attribute when `data.attributes.global` is `false`.                                                                                      |                                                                                                                                          |
```

File: website/docs/cloud-docs/api-docs/policies.mdx (hunk 1 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/policies.mdx]: fixed!

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 6]
```diff
@@ -797,6 +829,53 @@ curl \
   https://app.terraform.io/api/v2/policy-sets/polset-3yVQZvHzf5j3WRJ1/relationships/workspaces
 ```
 
+## Attach a Policy Set to exclusions
+
+`POST /policy-sets/:id/relationships/excludedWorkspaces`
```
Comment (ker-an) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:834]: Is the endpoint called `excludedWorkspaces` or `excluded`?

File: website/docs/cloud-docs/api-docs/policy-sets.mdx (hunk 4 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:4]: @rkoron007

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 7]
```diff
@@ -40,7 +40,7 @@ description: >-
 
 [Policy Enforcement](/terraform/cloud-docs/policy-enforcement) lets you use the policy-as-code frameworks Sentinel and Open Policy Agent (OPA) to apply policy checks to Terraform Cloud workspaces.
 
-[Policy sets](/terraform/cloud-docs/policy-enforcement/manage-policy-sets) are collections of policies that you can apply globally or to specific [projects](/terraform/cloud-docs/workspaces/organize-workspaces-with-projects) and workspaces. For each run in the selected workspaces, Terraform Cloud checks the Terraform plan against the policy set.
+[Policy sets](/terraform/cloud-docs/policy-enforcement/manage-policy-sets) are collections of policies that you can apply globally or to specific [projects](/terraform/cloud-docs/workspaces/organize-workspaces-with-projects) and workspaces, except the ones in the exclusions list. For each run in the selected workspaces, Terraform Cloud checks the Terraform plan against the policy set.
```
[the remark this answers is not in the document]
Comment (rkoron007) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:43]: I agree with Keran, I don't think we should mention exclusions here because we are still explaining what a policy set is. We want folks to have that firm understanding before building on that knowledge:
```suggestion
[Policy sets](/terraform/cloud-docs/policy-enforcement/manage-policy-sets) are collections of policies that you can apply globally or to specific [projects](/terraform/cloud-docs/workspaces/organize-workspaces-with-projects) and workspaces. For each run in the selected workspaces, Terraform Cloud checks the Terraform plan against the policy set.
```

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 8]
```diff
@@ -90,11 +90,12 @@ Properties without a default value are required.
...
+| `data.relationships.exclusions.data[]`        | array\[object] | `[]`      | A list of resource identifier objects that defines which excluded workspaces and are associated with the policy set. These objects must contain `id` and `type` properties, and the `type` property must be `workspaces`. For example, `{ "id": "ws-FVVvzCDaykN1oHiw", "type": "workspaces" }`.                                                                                                                                                                                                                                               |
```
[the remark this answers is not in the document]
Comment (rkoron007) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:98]: Adding to Keran's above suggestion with a teeny tweak for active voice:
```suggestion
| `data.relationships.exclusions.data[]`        | array\[object] | `[]`      | A list of resource identifier objects specifying which workspaces Terraform Cloud excludes from a policy set's enforcement. These objects must contain `id` and `type` properties, and the `type` property must be `workspaces`. For example, `{ "id": "ws-FVVvzCDaykN1oHiw", "type": "workspaces" }`.                                                                                                                                                                                                                                               |
```

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 9]
```diff
@@ -533,7 +554,7 @@ curl --request GET \
 }
 ```
 
--> **Note:** The `data.relationships.projects` and `data.relationships.workspaces` objects refer to projects and workspaces directly attached to the policy set. These keys are omitted for policy sets marked as global, which are implicitly related to all of the organization's workspaces.
+-> **Note:** The `data.relationships.projects`, `data.relationships.workspaces`, and `data.relationships.exclusions` objects refer to projects, workspaces and exclusions directly attached to the policy set. These keys are omitted for policy sets marked as global, which are implicitly related to all of the organization's workspaces.
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:557]: Can we split this out into two sentences? I think `exclusions` are different enough that they need more explanation:
```suggestion
-> **Note:** The `data.relationships.projects` and `data.relationships.workspaces` refer to the projects and workspaces attached to the policy set. The `data.relationships.exclusions` object refers to the attached workspaces that Terraform Cloud excludes from a policy set's rules. Terraform Clouds omits these keys for policy sets marked as global, which are implicitly related to all of the organization's workspaces.
```

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 10]
```diff
@@ -568,7 +589,8 @@ Properties without a default value are required.
...
+| `data.relationships.workspaces`               | array\[object] | (previous value) | An array of references to workspaces that the policy set should be assigned to. Sending an empty array clears all workspace assignments. You can only specify this attribute when `data.attributes.global` is `false`.                                                                                       | 
+| `data.relationships.exclusions`               | array\[object] | (previous value) | An array of references to workspaces that the policy set should not be assigned to. Sending an empty array clears all exclusions assignments. You can only specify this attribute when `data.attributes.global` is `false`.                                                                                      |                                                                                                                                          |
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:593]: Question: is it that Terraform Cloud doesn't _assign_ the policy set? Or that TFC doesn't _run_ the policy set for excluded workspaces? 

Small suggestion for active voice, but again not sure if this is correct without the above question:
```suggestion
| `data.relationships.exclusions`               | array\[object] | (previous value) | An array of references to workspaces that Terraform Cloud excludes from a policy set. Sending an empty array clears all exclusions assignments. You can only specify this attribute when `data.attributes.global` is `false`.                                                                                      |                                                                                                                                          |
```

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 11]
```diff
@@ -797,6 +829,53 @@ curl \
   https://app.terraform.io/api/v2/policy-sets/polset-3yVQZvHzf5j3WRJ1/relationships/workspaces
 ```
 
+## Attach a Policy Set to exclusions
+
+`POST /policy-sets/:id/relationships/excludedWorkspaces`
+
+| Parameter | Description                                                                                                 |
+| --------- | ----------------------------------------------------------------------------------------------------------- |
+| `:id`     | The ID of the policy set to attach to excluded workspaces. Use the "List Policy Sets" endpoint to find IDs. |
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:838]: The phrasing here is a little hard to follow. Is the idea that you pass Terraform Cloud the id of a policy set _and_ the workspaces it should exclude from that set? I tried rephrasing, but I'm not sure if this is accurate?
```suggestion
| `:id`     | The ID of a policy set that you want Terraform Cloud to exclude from the workspaces you specify. Use the "List Policy Sets" endpoint to find IDs. |
```

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 12]
```diff
@@ -1123,8 +1250,9 @@ The following relationships may be present in various responses for policy sets:
...
 | `policies`        | Individually managed policies which are associated with the policy set.                                                                                                                        |
-| `projects`        | The project this policy set applies to.                                                                                                                                                  |
-| `workspaces`      | The workspaces to which the policy set applies.                                                                                                                                                |
+| `projects`        | The projects this policy set applies to.                                                                                                                                                  |
+| `workspaces`      | The workspaces this policy set applies to. 
+| `exclusions`      | The workspaces excluded from to this policy set's enforcement.                                                                                                                                                 |
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:1255]:
```suggestion
| `exclusions`      | The workspaces excluded from a policy set's enforcement.                                                                                                                                                 |
```

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 13]
```diff
@@ -941,6 +1020,54 @@ curl \
   https://app.terraform.io/api/v2/policy-sets/polset-3yVQZvHzf5j3WRJ1/relationships/workspaces
 ```
 
+## Detach the Policy Set from exclusions
+
+`DELETE /policy-sets/:id/relationships/excludedWorkspaces`
+
+| Parameter | Description                                                                                                   |
+| --------- | ------------------------------------------------------------------------------------------------------------- |
+| `:id`     | The ID of the policy set to detach from excluded workspaces. Use the "List Policy Sets" endpoint to find IDs. |
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:1029]: The negative phrasing is a bit confusing here. This effectively just returns a policy set's enforcement back to normal state, right? Can you please add something to that effect?

**Review by rkoron007 — APPROVED** [review]
This is such a cool feature! Thank you for writing these docs, left some questions and feedback! Please feel free to push back on anything that doesn't feel good. ✨

File: website/docs/cloud-docs/api-docs/policy-sets.mdx (hunk 6 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx]: it's exclusions now, fixed!

File: website/docs/cloud-docs/api-docs/policy-sets.mdx (hunk 13 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:1029]: @rkoron007 would this be better: The ID of the policy set reattach to excluded workspaces.

File: website/docs/cloud-docs/api-docs/policy-sets.mdx (hunk 10 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:593]: Its this one: Or that TFC doesn't run the policy set for excluded workspaces

Comment (rkoron007) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:593]: Thank you for clarifying! With that background,  I think this feels a little more clear (feel free to push back if you don't think so though!):
```suggestion
| `data.relationships.exclusions`               | array\[object] | (previous value) | An array of references to excluded workspaces that Terraform Cloud will not enforce this policy set upon. Sending an empty array clears all exclusions assignments. You can only specify this attribute when `data.attributes.global` is `false`.                                                                                      |                                                                                                                                          |
```

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 14]
```diff
@@ -797,6 +829,53 @@ curl \
   https://app.terraform.io/api/v2/policy-sets/polset-3yVQZvHzf5j3WRJ1/relationships/workspaces
 ```
 
+## Attach a Policy Set to exclusions
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:832]:
```suggestion
## Exclude a workspace from a policy set 
```

Comment (rkoron007) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:832]: On further thought, I think we should use "exclude"/ "reinclude" here to explain how a workspace is "Excluded from a policy set's enforcement", which can be undone by "reincluding a workspace to a policy set's enforcement".  What do you think?

Comment (rkoron007) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:832]: Or does this both "attach" and "exclude"? Sorry to nitpick, negative phrasing is so confusing by default 😅

File: website/docs/cloud-docs/api-docs/policy-sets.mdx (hunk 11 above)
Comment (rkoron007) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:838]: (gentle bump on this, do you think the above suggestion works?)

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 15]
```diff
@@ -941,6 +1020,54 @@ curl \
   https://app.terraform.io/api/v2/policy-sets/polset-3yVQZvHzf5j3WRJ1/relationships/workspaces
 ```
 
+## Detach the Policy Set from exclusions
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:1023]:
```suggestion
## Reinclude a workspace to a policy set
```

File: website/docs/cloud-docs/api-docs/policy-sets.mdx (hunk 13 above)
Comment (rkoron007) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:1029]: I think the verb "reattach" feels confusing here. It's excluded and reincluded, right?

```suggestion
| `:id`     | The ID of the policy set Terraform Cloud should reinclude (enforce) on the specified workspaces. Use the "List Policy Sets" endpoint to find IDs. |
```

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 16]
```diff
@@ -941,6 +1020,54 @@ curl \
...
+| -------- | -------------- | ------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `data[]` | array\[object] |         | A list of resource identifier objects that defines which excluded workspaces the policy set will be detached from. These objects must contain `id` and `type` properties, and the `type` property must be `workspaces` (e.g. `{ "id": "ws-2HRvNs49EWPjDqT1", "type": "workspaces" }`). Obtain workspace IDs from the [workspace settings](/terraform/cloud-docs/workspaces/settings) or the [Show Workspace](/terraform/cloud-docs/api-docs/workspaces#show-workspace) endpoint. |
```
Comment (rkoron007) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:1047]:
```suggestion
| `data[]` | array\[object] |         | A list of resource identifier objects that defines which workspaces Terraform Cloud should reinclude (enforce) this policy set on. These objects must contain `id` and `type` properties, and the `type` property must be `workspaces` (e.g. `{ "id": "ws-2HRvNs49EWPjDqT1", "type": "workspaces" }`). Obtain workspace IDs from the [workspace settings](/terraform/cloud-docs/workspaces/settings) or the [Show Workspace](/terraform/cloud-docs/api-docs/workspaces#show-workspace) endpoint. |
```

**Review by rkoron007 — COMMENTED** [review]
Added a few more comments, please feel free to push back on anything that doesn't make sense though! These are looking good ✨

File: website/docs/cloud-docs/api-docs/policy-sets.mdx (hunk 10 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx]: I have changed it to: An array of references to excluded workspaces that the policy set should be assigned to.

Is this better?

File: website/docs/cloud-docs/api-docs/policy-sets.mdx (hunk 14 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:832]: Exclude sounds good!

File: website/docs/cloud-docs/api-docs/policy-sets.mdx (hunk 11 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:838]: This is accurate, but the wording can be perhaps simplified?
How about this: The ID of a policy set that you want to exclude the specified workspaces from. What do you think about this?

File: website/docs/cloud-docs/api-docs/policy-sets.mdx [hunk 17]
```diff
@@ -701,7 +733,7 @@ curl \
   https://app.terraform.io/api/v2/policy-sets/polset-3yVQZvHzf5j3WRJ1/relationships/policies
 ```
 
-## Attach a Policy Set to projects
+## Attach a Policy Set to Projects
```
Comment (Netra2104) [inline website/docs/cloud-docs/api-docs/policy-sets.mdx:736]: @rkoron007 fixed Capitalization of the titles to make them consistent.

Comment (rkoron007) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:736]: Thank you so much for putting in the effort here.  We've recently [updated the style guide](https://github.com/hashicorp/engineering-docs/blob/main/writing/style-guide.md#titles-and-headers) to specify that headers should use sentence casing. It's a not a huge priority, but it you are willing to make the change across the page to ensure all headings use sentence casing instead that would be **amazing**! No problem if you'd rather not though, just let me know and I can handle it later!

We also don't capitalize feature names, just _product_ names (so we capitalize Terraform Cloud, but not projects which are a _feature_ of Terraform Cloud). So we'd ideally update this to:
```suggestion
## Attach a policy set to projects
```

File: website/docs/cloud-docs/api-docs/policy-sets.mdx (hunk 11 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:838]: @rkoron007 in case you missed it, any thoughts on this?

File: website/docs/cloud-docs/api-docs/policy-sets.mdx (hunk 17 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx]: I have fixed it in all places on this page.

File: website/docs/cloud-docs/api-docs/policy-sets.mdx (hunk 11 above)
Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx:838]: On second thought, I think this works.

Comment (Netra2104) [reply website/docs/cloud-docs/api-docs/policy-sets.mdx]: fixed!

**Review by rkoron007 — APPROVED** [review]
This looks great to me! Thank you so much for going above and beyond on this page! ✨ 🥇
